salt/modules/nilrt_ip.py

```diff
@@ -371,14 +370,12 @@ def _get_static_info(interface):
         if information['IPv4.method'] == 'manual' and information['IPv4'] != '':
             ipv4_information = information['IPv4'].split('/')
             data['ipv4']['address'] = ipv4_information[0]
-            data['ipv4']['dns'] = information['Nameservers'].split(',')
+            data['ipv4']['dns'] = '' if information['Nameservers'] == '\'\'' else information['Nameservers'].split(',')
             data['ipv4']['netmask'] = ipv4_information[1]
             data['ipv4']['gateway'] = ipv4_information[2]
             data['ipv4']['requestmode'] = 'static'
-        else:
-            information = _load_config('General', ['NetworkInterfaceBlacklist'], filename=CONNMAN_MAIN_CONFIG)
-            if data['label'] in information['NetworkInterfaceBlacklist']:
-                data['ipv4']['requestmode'] = 'disabled'
+        elif information['IPv4'] == 'off':
```

Comment:
Yes, it makes sense, because if an interface is not off (disabled) or in static mode then it means it's using dhcp mode, and data['ipv4']['requestmode'] is by default `dhcp_linklocal_all`. This function is called only when an interface is not connected to a cable so even if it's using dhcp mode, the interface won't have any IP address.